### DISCUSSION TITLE
How to set up Waveshare RS485 to ETH B using Mac?

### DISCUSSION THREAD

**Discussion by ywteh** [nathanmarlor/foxess_modbus#665] [Q&A] [answered]
Hi,

So I went and got the Waveshare RS485 to ETH B to try to connect home assistant to my FoxESS Inverter. But then I realised that Vircom is a Window software and I don't have a Windows machine. Is there another way to set up the device using Mac?  I also tried to find the device on my network and connect to it using web browser (I presume this is what the wiki meant) but can't find it either on my network. Any advice?

cheers,
-yw

**Comment by ywteh** [comment]
I managed to try vircom on windows and it couldn't find the device either. I *think* it's connected (power light on, ethernet light green) but can't seem to find it on network it seems.

Sorry for the newbie questions...

Thank you.

**Reply by FozzieUK** [reply, marked as answer]
Vircom should be able to find it, it might be worth factory resetting the device as it has been known to not respond, particularly if the firmware has been upgraded see here https://github.com/nathanmarlor/foxess_modbus/wiki/Waveshare-RS485-to-ETH-%28B%29#reset-the-device-restore-factory-settings

Another thing to check is the waveshare is a relatively slow device on ethernet, if your router supports it, it is better to turn off auto-negotiation on that port and set it to 100mb duplex.

**Comment by ywteh** [comment]
Thank you!  A lot of patience and a bit of fiddling later, I managed to connect to the device and set it up, following instructions. Thank you!

**Reply by FozzieUK** [reply]
Thanks, glad you have sorted it 👍
